Add readme and package.json files to the button toggle

iveysaur opened the pull request:
Add `README.md` and `package.json` for button toggle.

R: @jelbourn @kara 
CC: @hansl @robertmesserle

kara commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,97 @@
+# MdButtonToggle
+
+`MdButtonToggle` is a group of buttons that can be toggled. There are two modes, `multiple` and
+`exclusive`. `Multiple` is a group in which multiple buttons can be toggled at once. `Exclusive` is
```
I think it might be clearer if you say "selected" here, rather than "toggled".  Technically all of them can be toggled in any mode, but only one (or multiple) can be selected.

kara commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,97 @@
+
+## `<md-button-toggle>`
+
+### Bound Properties
```
I'd put the summary of properties and events at the bottom. It's usually more useful for people to see basic usage examples first.

kara commented on src/components/button-toggle/README.md:
````diff
@@ -0,0 +1,97 @@
+
+```html
+<md-button-toggle>Bold</md-button-toggle>
+```
````
Can you add a screenshot of what this might look like?

kara replied:
This looks good, but I think it's showing two button toggles? Does the code match the image?

kara commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,97 @@
+    <md-button-toggle>Eggs</md-button-toggle>
+    <md-button-toggle>Sugar</md-button-toggle>
+    <md-button-toggle>Milk</md-button-toggle>
+</md-button-toggle-group>
```
Can you add screenshots for this and exclusive selection as well?

kara replied:
Nitpick: It might be a little confusing that you are showing a disabled button-toggle along with a selected one.  It almost looks like multi-select.  Perhaps create another section for the "disabled" example, so it's clear?

kara commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,97 @@
+### Multiple Selection
+
+`md-button-toggle` can be used in a multiple selection group when surrounded by
+`md-button-toggle-group multiple`. This styles all buttons within the group to appear as a single
```
Hmm, have you thought about putting the exclusive selection section before multiple selection? It's probably more common, and that way, you can explain `md-button-toggle-group` without the `multiple` attribute first.

kara commented on src/components/button-toggle/README.md:
````diff
@@ -0,0 +1,97 @@
+```
+
+## Dynamic Exclusive Selection
+
````
What is dynamic exclusive selection?  Probably need some text here.

kara commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,97 @@
+# MdButtonToggle
+
+`MdButtonToggle` is a group of buttons that can be toggled. There are two modes, `multiple` and
```
You might want to take a step back and start with importing MD_BUTTON_TOGGLE_DIRECTIVES. Definitely mention that you need `MdUniqueSelectionDispatcher` and give an example of that.

iveysaur commented:
@kara: addressed comments aside from images which will be added tomorrow.

Does the order look better? I'm also not entirely sure I did a good job at explaining the dynamic section. I also kept the paragraph about multiple and selection in the same place because they were just floating around if I moved them after the setup, is that okay?

iveysaur commented:
@kara: addressed comments

jelbourn commented on src/components/button-toggle/README.md:
````diff
@@ -0,0 +1,155 @@
+### Setup
+
+Importing the symbols:
+```typescript
````
nit: you can put `ts` instead of writing out `typescript`

jelbourn commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,155 @@
+`MdButtonToggle` is a group of buttons that can be toggled.
+There are two modes, `multiple` and `exclusive`.
+When in 'multiple' mode, multiple buttons can be selected at once.
+When in 'exclusive'mode, only one button can be selected at a time.
```
For these sentences, I would add:
```
When in 'multiple' mode, multiple buttons can be selected at once (like checkboxes).
When in 'exclusive'mode, only one button can be selected at a time (like radio buttons).
```

iveysaur commented:
@jelbourn: Addressed comments.

kara commented on src/components/button-toggle/README.md:
```diff
@@ -0,0 +1,155 @@
+
+`MdButtonToggle` is a group of buttons that can be toggled.
+There are two modes, `multiple` and `exclusive`.
+When in 'exclusive'mode, only one button can be selected at a time (like radio buttons).
```
missing a space between 'exclusive' and 'mode'

kara commented:
LGTM once you fix that space :)

jelbourn commented:
LGTM
